### ISSUE TITLE
Scylla crashed on SIGSEGV when overloaded with a read workload

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#1771]
*Installation details*
Scylla version (or git commit hash): 1.4-rc2

journactl
```
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 6] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 4] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 7] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 4] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 4] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 6] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 1] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 3] seastar - Exceptional future ignored: semaphore_timed_out (Semaphore timedout)
lines 5435-5485/510338 1%
.
.
.
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 7] seastar - Exceptional future ignored: semaphore_timed_out (Semaphore timedout)
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 7] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: Segmentation fault on shard 5.
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: Backtrace:
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00000000004e2dea
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00000000004e2e93
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00007f3ee94600ff
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00000000005b3040
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00000000005d194c
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x000000000110eedb
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x000000000111781c
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x0000000001117e83
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00000000004e4a4e
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x000000000052f7f2
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x000000000053b7f8
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x000000000056f65d
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00007f3ee9458dc4
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0x00007f3ee918621c
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 2] seastar - Exceptional future ignored: semaphore_timed_out (Semaphore timedout)
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 7] storage_proxy - Exception when communicating with 172.30.0.203: Semaphore timedout
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]:  [shard 7] seastar - Exceptional future ignored: semaphore_timed_out (Semaphore timedout)
Oct 19 16:30:11 ip-172-30-0-203 scylla[30723]: 0xffffffffffffffff
```

coredump info

```
[root@ip-172-30-0-203 centos]# sudo coredumpctl info 30723
           PID: 30723 (scylla)
           UID: 994 (scylla)
           GID: 1001 (scylla)
        Signal: 11 (SEGV)
     Timestamp: Wed 2016-10-19 16:30:11 UTC (4h 43min ago)
  Command Line: /usr/bin/scylla --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --collectd-address=127.0.0.1:25826 --collectd=1 --collectd-poll-period 3000 --network-stack posix --num-io-queues 8 --
    Executable: /usr/bin/scylla
 Control Group: /system.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: system.slice
       Boot ID: cadb793725514780995acceb4f3c3b06
    Machine ID: 8ea3e6ea03b64d018847fcdf669935d4
      Hostname: ip-172-30-0-203
       Message: Process 30723 (scylla) of user 994 dumped core.
                
                Stack trace of thread 30728:
                #0  0x00000000005b3041 _ZN9file_desc8shutdownEi (scylla)
                #1  0x00000000005d194d _ZN16connected_socket15shutdown_outputEv (scylla)
                #2  0x000000000110eedc _ZN3rpc8protocolIN3net10serializerENS1_14messaging_verbEE10connection14stop_send_loopEv (scylla)
                #3  0x000000000111781d _ZZN3rpc8protocolIN3net10serializerENS1_14messaging_verbEE6server10connection7processEvENKUl6futureIJEEE0_clES8_ (scylla)
                #4  0x0000000001117e84 _ZZN6futureIJEE12then_wrappedIZN3rpc8protocolIN3net10serializerENS4_14messaging_verbEE6server10connection7processEvEUlS0_E0_S0_EET0_OT_ENUlSD_E_clI12future_stateIJEEEEDaSD_
                #5  0x00000000004e4a4f _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                #6  0x000000000052f7f3 _ZN7reactor3runEv (scylla)
                #7  0x000000000053b7f9 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.3327 (scylla)
                #8  0x000000000056f65e _ZNKSt8functionIFvvEEclEv (scylla)
                #9  0x00007f3ee9458dc5 start_thread (libpthread.so.0)
                #10 0x00007f3ee918621d __clone (libc.so.6)
                
                Stack trace of thread 30729:
                #0  0x0000000000571500 _ZN6memory10small_pool8allocateEv (scylla)
                #1  0x000000000057218e _ZN6memory8allocateEm (scylla)
                #2  0x0000000000572295 _Znam (scylla)
                #3  0x0000000001003eab _ZN9__gnu_cxx13new_allocatorI17nonwrapping_rangeI21clustering_key_prefixEE8allocateEmPKv (scylla)
                #4  0x0000000000fff8f5 _ZN5query15partition_sliceC2ERKS0_ (scylla)
                #5  0x00000000012da8ef _ZN5query12read_commandC4ERKS0_ (scylla)
                #6  0x00000000012d6e0a _ZZN7service5pager12query_pagers4impl10fetch_pageEjNSt6chrono10time_pointI8db_clockNS3_8durationIlSt5ratioILl1ELl1000EEEEEEENKUlRT_E_clIN4cql39selection18result_set_builder
                #7  0x0000000000bf93cc _ZN4cql310statements16select_statement7executeERN7seastar7shardedIN7service13storage_proxyEEERNS4_11query_stateERKNS_13query_optionsE (scylla)
                #8  0x0000000000db9a50 operator() (scylla)
                #9  0x0000000000dba75d then<cql3::query_processor::process_statement(shared_ptr<cql3::cql_statement>, service::query_state&, const cql3::query_options&)::<lambda()>, future<shared_ptr<transport::
                #10 0x0000000000b34968 _ZN9transport10cql_server10connection15process_executeEtNSt12experimental15fundamentals_v117basic_string_viewIaSt11char_traitsIaEEEN7service12client_stateE (scylla)
                #11 0x0000000000b399f8 _ZZN9transport10cql_server10connection19process_request_oneENSt12experimental15fundamentals_v117basic_string_viewIaSt11char_traitsIaEEEhtN7service12client_stateENS1_20traci
                #12 0x0000000000b3aa83 apply (scylla)
                #13 0x0000000000b3c933 _ZZZZZZN9transport10cql_server10connection15process_requestEvENKUlO6futureIJNSt12experimental15fundamentals_v18optionalINS_19cql_binary_frame_v3EEEEEE_clES9_ENKUl15semaphor
                #14 0x0000000000b3e260 apply<transport::cql_server::connection::process_request()::<lambda(future<std::experimental::fundamentals_v1::optional<transport::cql_binary_frame_v3> >&&)>::<lambda(semap
                #15 0x0000000000b406fd then<transport::cql_server::connection::process_request()::<lambda(future<std::experimental::fundamentals_v1::optional<transport::cql_binary_frame_v3> >&&)>::<lambda(semaph
                #16 0x0000000000b415ba then<transport::cql_server::connection::process_request()::<lambda(future<std::experimental::fundamentals_v1::optional<transport::cql_binary_frame_v3> >&&)>::<lambda(semaph
                #17 0x0000000000b436ae operator()<future_state<std::experimental::fundamentals_v1::optional<transport::cql_binary_frame_v3> > > (scylla)
                #18 0x00000000004e4a4f _ZN7reactor9run_tasksER15circular_buf
```

**Comment by kongove** [comment]
Attached coredump logs from SCT test (performance_regression_test.py: test_read)
I tested 3 times of read test, they all have crash, there are two nodes crashed in second test.

@tgrabiec can you help to check if it's same issue?

[coredump-read-repeat1-node1.txt](https://github.com/scylladb/scylla/files/540353/coredump-read-repeat1-node1.txt)
[coredump-read-repeat3-node2.txt](https://github.com/scylladb/scylla/files/540354/coredump-read-repeat3-node2.txt)
[coredump-read-repeat3-node3.txt](https://github.com/scylladb/scylla/files/540355/coredump-read-repeat3-node3.txt)
[coredump-read-repeat2-node3.txt](https://github.com/scylladb/scylla/files/540356/coredump-read-repeat2-node3.txt)

**Comment by tgrabiec** [comment]
I opened a seastar issue for the crash during connection closing on error: https://github.com/scylladb/seastar/issues/210

**Comment by tgrabiec** [comment]
Fixed by e8edaaf6a48172d768c951d4b28eba8757ce6a29

**Comment by kongove** [comment]
On Tue, Oct 25, 2016 at 11:47 PM, Tomasz Grabiec <notifications@github.com>
wrote:

> Fixed by e8edaaf
> <https://github.com/scylladb/scylla/commit/e8edaaf6a48172d768c951d4b28eba8757ce6a29>
>

This fix need to be merged into 1.4 branch. When can we have a new RC build
with this fix?

**Comment by tgrabiec** [comment]
@kongove It was merged to 1.4 in https://github.com/scylladb/scylla/commit/1c278d9abfefbf4e921b5d5ea501322de592ba05 . 

About the RC build, \cc @penberg @slivne

**Comment by penberg** [comment]
@tgrabiec @kongove I managed to screw up the -rc3 build yesterday. There's a new one building as we speak. It should be ready by the end of today.
